cpp/src/io/json/reader_impl.cu

```diff
@@ -156,9 +156,9 @@ std::unique_ptr<table> create_json_keys_info_table(
   info_columns.emplace_back(make_numeric_column(data_type(type_id::UINT16), num_keys));
   info_columns.emplace_back(make_numeric_column(data_type(type_id::UINT32), num_keys));
   // Create a table out of these columns to pass them around more easily
-  auto info_table = std::make_unique<table>(std::move(info_columns));
-  rmm::device_scalar<mutable_table_device_view> info_table_mdv(
-    *mutable_table_device_view::create(info_table->mutable_view(), stream), stream);
+  auto info_table           = std::make_unique<table>(std::move(info_columns));
+  auto const info_table_mdv = mutable_table_device_view::create(info_table->mutable_view(), stream);
+  rmm::device_scalar<mutable_table_device_view> d_info_table_mdv(*info_table_mdv, stream);
```

Comment:
Why is this creating a `device_scalar<mutable_table_device_view>`? That sounds really odd to me. `mutable_table_device_view` should be passed by value into kernels.